x-pack/plugins/cases/server/common/utils.ts

```diff
@@ -394,3 +404,50 @@ export const asArray = <T>(field?: T | T[] | null): T[] => {
 export const assertUnreachable = (x: never): never => {
   throw new Error('You should not reach this part of code');
 };
+
+export const getApplicationRoute = (
+  appRouteInfo: { [K in keyof typeof OWNER_INFO]: { appRoute: string } },
+  owner: string
+): string => {
+  const appRoute = isValidOwner(owner)
+    ? appRouteInfo[owner].appRoute
+    : OWNER_INFO[GENERAL_CASES_OWNER].appRoute;
+
+  return appRoute.startsWith('/') ? appRoute : `/${appRoute}`;
+};
+
+export const getCaseViewPath = (params: {
+  publicBaseUrl: NonNullable<IBasePath['publicBaseUrl']>;
+  caseId: string;
+  owner: string;
+  commentId?: string;
+  tabId?: CASE_VIEW_PAGE_TABS;
+}): string => {
+  const normalizePath = (path: string): string => path.replaceAll('//', '/');
```

Comment:
If we used the URL class would we be able to avoid removing trailing slashing and the normalizing?

Reply:
Unfortunately, the URL class keeps the double slashes. For example:

```
node > new URL('https://e.com//api//cases')
URL {
  href: 'https://e.com//api//cases',
  origin: 'https://e.com',
  protocol: 'https:',
  username: '',
  password: '',
  host: 'e.com',
  hostname: 'e.com',
  port: '',
  pathname: '//api//cases',
  search: '',
  searchParams: URLSearchParams {},
  hash: ''
}
```